app/controllers/v0/sign_in_controller.rb

```diff
@@ -175,31 +180,36 @@ def revoke_all_sessions
 
     def logout # rubocop:disable Metrics/MethodLength
       client_id = params[:client_id].presence
-      anti_csrf_token = anti_csrf_token_param.presence
-
-      if client_config(client_id).blank?
+      if client_id && client_config(client_id).blank?
         raise SignIn::Errors::MalformedParamsError.new message: 'Client id is not valid'
       end
 
-      unless load_user(skip_expiration_check: true)
-        raise SignIn::Errors::LogoutAuthorizationError.new message: 'Unable to Authorize User'
+      if (session = SignIn::OAuthSession.find_by(handle: @access_token&.session_handle))
```

Comment:
could we follow the same format as before but with the new function? `unless access_token_optional_authenticate`, and then just throw an error that's caught if the user isn't authenticated? Or is there a reason the general flow needed to be updated?

Reply:
The new `if/else` flow is to make the method work with or without an active access_token. If there's a valid session it behaves normally, if there isn't we still clear cookies, log the failure to find the session, etc. Should I move the `else` block into a new exception handling?